### PR TITLE
docs: Update 'user guides' -> 'client apps'

### PR DESCRIPTION
**Pull request by jamilbk**
- This terminology / naming makes things more clear for most admins/users.
- Fixes layout of /kb/client-apps to align better

### CODE REVIEW

File: rust/gui-client/docs/intended_behavior.md [hunk 1]
```diff
@@ -40,13 +40,13 @@ Best performed on a clean VM
 1. Sign out of Firezone without quitting
 1. Check the IP again, expect your own IP (Checks for regressions in https://github.com/firezone/firezone/pull/5828)
 1. Export the logs
-1. Expect the zip file to start with "firezone_logs_"
+1. Expect the zip file to start with "firezone*logs*"
```
Comment (ReactorScram) [inline rust/gui-client/docs/intended_behavior.md]:
```suggestion
1. Expect the zip file to start with `firezone_logs_`
```
Prettifier kicked in again?

File: website/src/app/kb/deploy/users/readme.mdx [hunk 2]
```diff
@@ -20,13 +20,14 @@ as the actor type.
...
 Users with the `User` role can only sign in to the Firezone Clients to access
-Resources, while users with the `Admin` can sign in to the admin portal to
+Resources, while users with the `Admin` role can sign in to the admin portal to
 perform administrative tasks as well as sign in to Firezone Clients to access
 Resources.
 
-Optionally, you can choose to assign the user to one or more groups. This will
-ultimately determine the policies that apply to the user, so it's important to
-ensure that the groups have the correct policies defined.
+After creating a user, you can choose to assign the user to one or more groups.
```
Comment (ReactorScram) [inline website/src/app/kb/deploy/users/readme.mdx]:
```suggestion
After creating a user, you can assign the user to one or more groups.
```

File: website/src/app/kb/deploy/users/readme.mdx [hunk 3]
```diff
@@ -20,13 +20,14 @@ as the actor type.
...
+This will ultimately determine the policies that apply to the user, so it's
```
Comment (ReactorScram) [inline website/src/app/kb/deploy/users/readme.mdx]:
```suggestion
Groups determine the policies that apply to the user, so it's
```

File: website/src/app/kb/readme.mdx [hunk 4]
```diff
@@ -29,9 +29,8 @@ Use the links below to get started.
 - [Administer](/kb/administer): Everything related to typical, day-to-day
   management of your deployment.
 - [Automate](/kb/automate): Deploy with automation tools like Terraform.
-- [End-user guides](/kb/user-guides): Instructions you can pass along to
-  Firezone end-users, such as installing Clients, using them, and related
-  troubleshooting tips.
+- [End-user guides](/kb/client-apps): Documentation for installing and using the
```
Comment (ReactorScram) [inline website/src/app/kb/readme.mdx]:
```suggestion
- [Client apps](/kb/client-apps): Documentation for installing and using the
```

File: website/src/components/Footer/index.tsx [hunk 5]
```diff
@@ -231,10 +231,10 @@ export default function Footer() {
             </Link>
           </div>
           <div className="flex p-2 items-center justify-center space-x-5">
-            <AppleIcon size={5} href="/kb/user-guides/macos-client" />
-            <WindowsIcon size={5} href="/kb/user-guides/windows-client" />
-            <LinuxIcon size={5} href="/kb/user-guides/linux-gui-client" />
-            <AndroidIcon size={5} href="/kb/user-guides/android-client" />
+            <AppleIcon size={5} href="/kb/client-apps/macos-client" />
+            <WindowsIcon size={5} href="/kb/client-apps/windows-client" />
+            <LinuxIcon size={5} href="/kb/client-apps/linux-gui-client" />
```
Comment (ReactorScram) [inline website/src/components/Footer/index.tsx:236]: Looks like some places link to the Linux GUI Client (like here) but other places (like the email) link to the Linux CLI Client. Is this intentional?

File: website/src/app/kb/readme.mdx [hunk 6]
```diff
@@ -29,9 +29,8 @@ Use the links below to get started.
 - [Administer](/kb/administer): Everything related to typical, day-to-day
   management of your deployment.
 - [Automate](/kb/automate): Deploy with automation tools like Terraform.
-- [End-user guides](/kb/user-guides): Instructions you can pass along to
-  Firezone end-users, such as installing Clients, using them, and related
-  troubleshooting tips.
+- [Client apps](/kb/client-apps): Documentation for installing and using the
```
Comment (ReactorScram) [inline website/src/app/kb/readme.mdx:32]: Looks like this one got skipped, so I updated it

File: website/redirects.js [hunk 7]
```diff
@@ -79,4 +79,14 @@ module.exports = [
...
+  /*
+   * Redirects for old KB URLs
+   *
+   */
+  {
+    // TODO: Remove on or after 2024-10-21 after crawlers have re-indexed
+    source: "/kb/user-guides/:path",
+    destination: "/kb/client-apps/:path",
+    permanent: true,
+  },
```
Comment (ReactorScram) [inline website/redirects.js:91]: This alone doesn't cover `﻿﻿/kb/user-guides` with no trailing slash, which is the 2nd result on DuckDuckGo for "firezone user guides"

```suggestion
  {
    // TODO: Remove on or after 2024-10-21 after crawlers have re-indexed
    source: "/kb/user-guides/:path",
    destination: "/kb/client-apps/:path",
    permanent: true,
  },
  {
    source: "/kb/user-guides",
    destination: "/kb/client-apps",
    permanent: true,
  },
```

File: website/src/app/kb/client-apps/page.tsx [hunk 8]
```diff
@@ -2,7 +2,7 @@ import Content from "./readme.mdx";
 import { Metadata } from "next";
 
 export const metadata: Metadata = {
-  title: "User Guides • Firezone Docs",
+  title: "Client Apps • Firezone Docs",
```
Comment (ReactorScram) [inline website/src/app/kb/client-apps/page.tsx:5]: Did this one too

**Review by ReactorScram — CHANGES_REQUESTED** [review]
Makes sense but I found some small things that should probably change before it goes live

File: rust/gui-client/docs/intended_behavior.md (hunk 1 above)
Comment (jamilbk) [reply rust/gui-client/docs/intended_behavior.md]: Yeah its config changed in another PR so I guess it ran against all files again.

File: website/src/components/Footer/index.tsx (hunk 5 above)
Comment (jamilbk) [reply website/src/components/Footer/index.tsx:236]: The email needs to be updated, good catch. I'll do that now

File: website/src/app/kb/readme.mdx (hunk 6 above)
Comment (jamilbk) [reply website/src/app/kb/readme.mdx:32]: Good catch!

File: website/redirects.js (hunk 7 above)
Comment (jamilbk) [reply website/redirects.js:91]: Good catch I'll see if removing the last slash will achieve both

**Comment by jamilbk** [comment]
Redirects are working

File: website/redirects.js (hunk 7 above)
Comment (jamilbk) [reply website/redirects.js:91]: didn't look like it

**Comment by jamilbk** [comment]
@ReactorScram all feedback addressed

**Review by ReactorScram — APPROVED** [review]

**Pull request merged**